app/policies/application_policy.rb

```diff
@@ -13,10 +13,14 @@ def admin_user?
   end
 
   def regular_user?
-    @user.is_a?(Person)
+    @user.is_a?(Person) && @user.third_party_user? == false
   end
 
   def readonly_user?
-    @user.is_a?(ReadonlyUser)
+    @user.is_a?(ReadonlyUser) or @user.third_party_user? && @user.super_admin? == false
+  end
+
+  def third_party_admin_user?
+    @user.is_a?(Person) && @user.super_admin? && @user.third_party_user?
```

Comment:
If a third party admin is also a super admin, they will have the abilities of a super admin which I don't think you want.